src/api/worker/rest/EntityRestCache.ts

```diff
@@ -255,110 +243,186 @@ export class EntityRestCache implements IEntityRestCache {
+			|| typeModel.values._id.type !== ValueType.GeneratedId // we currently only store ranges for generated ids
+		) {
+			return this.entityRestClient.loadRange(typeRef, listId, start, count, reverse)
+		}
 
 		const range = await this.storage.getRangeForList(typeRef, listId)
 
-		if (range == null) {
-
-			// Create a new range and load everything
-			const entities = await this.entityRestClient.loadRange(typeRef, listId, start, count, reverse)
-
-			// Initialize a new range for this list
-			await this.storage.setNewRangeForList(typeRef, listId, start, start)
-
-			// The range bounds will be updated in here
-			return this.handleElementRangeResult(typeRef, listId, start, count, reverse, entities, count)
+		if (
+			range == null
+		) {
+			await this.createNewRange(typeRef, listId, start, count, reverse)
+		} else if (
+			!firstBiggerThanSecond(start, range.upper) && !firstBiggerThanSecond(range.lower, start)
+		) {
+			await this.extendFromWithinRange(typeRef, listId, start, count, reverse)
+		} else if (
+			(reverse && firstBiggerThanSecond(range.lower, start))
+			|| (!reverse && firstBiggerThanSecond(start, range.upper))
+		) {
+			await this.extendAwayFromRange(typeRef, listId, start, count, reverse)
 		} else {
+			await this.extendTowardsRange(typeRef, listId, start, count, reverse)
+		}
 
-			const startIsLocatedInRange = !firstBiggerThanSecond(start, range.upper) && !firstBiggerThanSecond(range.lower, start)
-
-			if (startIsLocatedInRange) {
-				const {newStart, newCount} = await this.recalculateRangeRequest(typeRef, listId, start, count, reverse)
-				if (newCount > 0) {
-					// We will be able to provide some entities from the cache, so we just want to load the remaining entities from the server
-					const entities = await this.entityRestClient.loadRange(typeRef, listId, newStart, newCount, reverse)
-					return this.handleElementRangeResult(typeRef, listId, start, count, reverse, entities, newCount)
-				} else {
-					// all elements are located in the cache.
-					return this.storage.provideFromRange(typeRef, listId, start, count, reverse)
-				}
-			} else {
+		return this.storage.provideFromRange(typeRef, listId, start, count, reverse)
+	}
 
-				const isLoadingAwayFromRange = (reverse && firstBiggerThanSecond(range.lower, start)) || (!reverse && firstBiggerThanSecond(start, range.upper))
+	/**
+	 * Creates a new list range, reading everything from the server that it can
+	 * range:         (none)
+	 * request:       *------>
+	 * range becomes: |------|
+	 * @private
+	 */
+	private async createNewRange<T extends ListElementEntity>(
+		typeRef: TypeRef<T>,
+		listId: Id,
+		start: Id,
+		count: number,
+		reverse: boolean
+	) {
+		// Create a new range and load everything
+		const entities = await this.entityRestClient.loadRange(typeRef, listId, start, count, reverse)
+
+		// Initialize a new range for this list
+		await this.storage.setNewRangeForList(typeRef, listId, start, start)
```

Comment:
I am a bit uneasy about creating a new range without elements and then populating it but it if it's empty it should be okay